src/main/java/com/google/devtools/build/lib/rules/java/JavaConfiguration.java

```diff
@@ -218,6 +223,58 @@ private static void checkLegacyToolchainFlagIsUnset(String flag, Label label)
     }
   }
 
+  private static void checkLanguageAndRuntimeVersion(@Nullable String rawLanguageVersion,
+      @Nullable String rawRuntimeVersion, boolean isExec) throws InvalidConfigurationException {
+    int languageVersion;
+    if (Strings.isNullOrEmpty(rawLanguageVersion)) {
+      return;
+    }
+    try {
+      languageVersion = Integer.parseUnsignedInt(rawLanguageVersion);
+    } catch (NumberFormatException e) {
+      // This is unexpected, but for the sake of compatibility with unusual toolchains, we should
+      // not fail the build.
+      return;
+    }
+
+    int runtimeVersion;
+    if (Strings.isNullOrEmpty(rawRuntimeVersion)) {
+      return;
+    }
+    int lastUnderscore = rawRuntimeVersion.lastIndexOf('_');
+    String rawRuntimeVersionLastPart;
+    if (lastUnderscore == -1) {
+      rawRuntimeVersionLastPart = rawRuntimeVersion;
+    } else {
+      rawRuntimeVersionLastPart = rawRuntimeVersion.substring(lastUnderscore + 1);
+    }
+    try {
+      runtimeVersion = Integer.parseUnsignedInt(rawRuntimeVersionLastPart);
+    } catch (NumberFormatException ignored) {
+      // Could be e.g. the "jdk" part of "local_jdk" without a version number, which we can't
+      // check at this point.
```

Comment:
It seems slightly dicey to figure out the Java features version from the string name of the runtime version. That convention doesn't hold for all of the Java runtimes we define at google, for example.

Is there somewhere we could do this validation when we have the `JavaRuntimeInfo`, in order to check the `java_runtime.version` instead of the value of `--java_runtime_version=`?